ckanext/dcatapchharvest/tests/test_dcatap_ch_parse.py

```diff
@@ -405,3 +405,21 @@ def test_multiple_rights_statements(self):
         p.parse(contents)
         dataset = [d for d in p.datasets()][0]
         resource = dataset["resources"][0]
+
+    def test_eu_themes_mapping(self):
+        contents = self._get_file_contents('catalog-themes.xml')
+        p = RDFParser(profiles=['swiss_dcat_ap'])
+        p.parse(contents)
+
+        for dataset in p.datasets():
+            eq_(
+                sorted(dataset['groups']),
+                [
+                    {'name': u'econ'},
+                    {'name': u'gove'},
+                    {'name': u'soci'},
+                ],
+                "Groups not mapped correctly for dataset {}".format(
+                    dataset['identifier']
+                )
+            )
```

Comment:
Could you please additionally test the mapping cases (Case 1, Case 2, Case 3) you provide?

Reply:
This test tests all of those cases. See the datasets in the fixture.